### PR TITLE
Add shapes endpoint

### PR DESCRIPTION
**Pull request by cedarbaum**
Add `ListShapes` and `GetShape` endpoints and also add shape references to `Trip` entities.

### CODE REVIEW

File: api/public.proto [hunk 1]
```diff
@@ -874,6 +931,7 @@ message Trip {
     Stop.Reference destination = 4;
     optional Vehicle.Reference vehicle = 5;
     bool direction_id = 6;
+    optional Shape.Reference shape = 7;
```
Comment (jamespfennell) [inline api/public.proto:934]: Do we need the Shape reference to be here? I understand this type already has a lot of stuff in it, but not sure the shape is important enough to be returned in reference types?

File: transiter.go [hunk 2]
```diff
@@ -196,6 +196,11 @@ func main() {
 						Usage: "Maximum number of vehicles that will be returned in a single list vehicles request. Specifying a value <= 0 will disable the limit.",
 						Value: 100,
 					},
+					&cli.Int64Flag{
```
Comment (jamespfennell) [inline transiter.go:199]: Would it be simpler to just have a max-entities-per-request flag and use the same value for vehicles, shapes, stops, etc?

File: api/public.proto (hunk 1 above)
Comment (cedarbaum) [reply api/public.proto]: I've removed this in the latest iteration. I realized shortly after I sent the PR that I never actually populated this field anywhere haha.

Comment (jamespfennell) [reply api/public.proto]: Nice!

File: transiter.go (hunk 2 above)
Comment (jamespfennell) [reply transiter.go]: Awesome thank you very much!

**Review by jamespfennell — APPROVED** [review]

**Pull request merged**